app/controllers/event_notes_controller.rb

```diff
@@ -0,0 +1,82 @@
+class EventNotesController < ApplicationController
+  include SerializeDataHelper
+
+  rescue_from Exceptions::EducatorNotAuthorized, with: :redirect_unauthorized!
+
+  before_action :authorize!, except: [:names]
+
+  def authorize!
+    student = Student.find(params[:student_id])
+    educator = current_educator
+    raise Exceptions::EducatorNotAuthorized unless educator.is_authorized_for_student(student)
+  end
+
+  def create
+    clean_params = params.require(:event_note).permit(*[
+      :student_id,
+      :event_note_type_id,
+      :text
+    ])
+
+    event_note = EventNote.new(clean_params.merge({
+      educator_id: current_educator.id,
+      recorded_at: Time.now
+    }))
+
+    if event_note.save
+      render json: serialize_event_note(event_note)
+    else
+      render json: { errors: event_note.errors.full_messages }, status: 422
+    end
+  end
+
+  def update
+    clean_params = params.require(:event_note).permit(*[
+      :id,
+      :student_id,
+      :event_note_type_id,
+      :text
+    ])
+
+    event_note_id = clean_params[:id]
+    event_note = EventNote.find(event_note_id)
+
+    # Save the state of the existing event note.
+    previous_event_note_revision = EventNoteRevision.where(
+      event_note_id: event_note_id
+    ).order(
+      :version
+    ).take
```

Comment:
I don't think `take` will respect your ordering and you'll end up with something slightly random depending on postgres (but most likely the smallest version number rather than the highest). Instead, I think the intention is to call `last`.